opent2t.p.thermostat/com.wink.thermostat/js/thingTranslator.js

```diff
@@ -0,0 +1,427 @@
+            break;
+        case 'awayMode':
+            desired_state['users_away'] = resourceSchema.modes[0] === 'away';
+            break;
+        case 'hvacMode': {
+                var mode = resourceSchema.modes[0];
+
+                if (mode === 'off') {
+                    desired_state['powered'] = false;
+                }
+                else {
+                    desired_state['powered'] = true;
+                    desired_state['mode'] = translatorHvacModeToDeviceHvacMode(mode);
+                }
+                break;
+            }
+        case 'targetTemperature':
+        case 'awayTemperatureHigh':
+        case 'awayTemperatureLow':
+        case 'fanTimerTimeout':
+        case 'fanMode':
+            throw new Error('NotImplemented');
+        default:
+            throw new Error('NotFound');
+    }
+
+    return result;
+}
+
+function validateResourceGet(resourceId) {
+    switch (resourceId) {
+        case 'humidity':
+        case 'awayTemperatureHigh':
+        case 'awayTemperatureLow':
+        case 'heatingFuelSource':
+        case 'fanTimerActive':
+        case 'fanTimerTimeout':
+        case 'fanMode':
+            throw new Error('NotImplemented');
+    }
+}
+
+function findResource(schema, di, resourceId) {
+    var entity = schema.entities.find((d) => {
+        return d.di === di;
+    });
+
+    if (!entity) {
+        throw new Error('NotFound');
+    }
+
+    var resource = entity.resources.find((r) => {
+        return r.id === resourceId;
+    });
+
+    if (!resource) {
+        throw new Error('NotFound');
+    }
+
+    return resource;
+}
+
+function getDeviceResource(translator, di, resourceId) {
+    validateResourceGet(resourceId);
+
+    return translator.get(true)
+        .then(response => {
+            return findResource(response, di, resourceId);
+        });
+}
+
+function postDeviceResource(di, resourceId, payload) {
+    if (di === deviceIds['opent2t.d.thermostat']) {
+        var putPayload = resourceSchemaToProviderSchema(resourceId, payload);
+
+        return winkHub.putDeviceDetailsAsync(deviceType, controlId, putPayload)
+            .then((response) => {
+                var schema = providerSchemaToPlatformSchema(response.data, true);
+
+                return findResource(schema, di, resourceId);
+            });
+    } else {
+        throw new Error('NotFound');
+    }
+}
+
+var controlId;
+var deviceType = 'thermostats';
+var winkHub;
+
+var deviceIds = {
+    'oic.d.thermostat': 'B610F482-19A4-4EC4-ADB3-3517C7969183',
+    'opent2t.d.thermostat': 'D5D37EB6-F428-41FA-AC5D-918F084A4C93'
+}
+
+// This translator class implements the 'org.opent2t.sample.thermostat.superpopular' schema.
+class Translator {
+
+    constructor(deviceInfo) {
+        console.log('Initializing device.');
+
+        validateArgumentType(deviceInfo, "deviceInfo", "object");
+
+        controlId = deviceInfo.deviceInfo.id;
```

Comment:
>deviceInfo.deviceInfo.id; [](start = 20, length = 25)

can you update this to be the same as the other translators? deviceInfo.deviceInfo.opent2t.controlId